modules/ensemble/lib/widget/slidable.dart

```diff
@@ -0,0 +1,225 @@
+import 'package:ensemble/framework/error_handling.dart';
+import 'package:ensemble/framework/extensions.dart';
+import 'package:ensemble/framework/scope.dart';
+import 'package:ensemble/framework/view/data_scope_widget.dart';
+import 'package:ensemble/widget/helpers/controllers.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter_slidable/flutter_slidable.dart';
+import 'package:ensemble/framework/widget/widget.dart';
+import 'package:ensemble/framework/action.dart';
+import 'package:ensemble/framework/event.dart';
+import 'package:ensemble/screen_controller.dart';
+import 'package:ensemble/util/utils.dart';
+import 'package:ensemble_ts_interpreter/invokables/invokable.dart';
+import 'package:ensemble/framework/model.dart';
+import 'package:ensemble/framework/widget/icon.dart' as iconframework;
+
+class EnsembleSlidable extends StatefulWidget
+    with
+        Invokable,
+        HasController<EnsembleSlidableController, EnsembleSlidableState> {
+  static const type = 'Slidable';
+
+  EnsembleSlidable({Key? key}) : super(key: key);
+
+  final EnsembleSlidableController _controller = EnsembleSlidableController();
+
+  @override
+  EnsembleSlidableController get controller => _controller;
+
+  @override
+  State<StatefulWidget> createState() => EnsembleSlidableState();
+
+  @override
+  Map<String, Function> getters() {
+    return {
+      'startContent': () => _controller.startContent,
```

Comment:
prob should mention this before - we need more descriptive name: startDrawer/endDrawer? @amin-nas

Reply:
Updated it to `startDrawer/endDrawer` for now, do let me if something better came up😅